### DISCUSSION TITLE
Determine if getStaticProps running at build time

### DISCUSSION THREAD

**Discussion by AsherFoster** [vercel/next.js#22036] [Help] [answered]
Because of the limitations I've mentioned here #21988, I'm wanting my app to do a client-side redirect during build-time, (`return { props: { redirectTo: url }, revalidate: 1 }`) and then immediately `revalidate` and return a server-side redirect (`return { redirect: { ... } }`)

However, I can't find a way to determine if getStaticProps is running at build-time or not. Is this possible?

Happy to accept a hack, as hopefully Next should support returning a redirect at build-time eventually.

**Comment by AsherFoster** [comment, marked as answer]
I've discovered `process.env.NEXT_PHASE`, which solves this

```ts
import { PHASE_PRODUCTION_BUILD } from 'next/constants';

// ...

if (process.env.NEXT_PHASE === PHASE_PRODUCTION_BUILD) {
  return {
    props: {
      redirectTo: href
    },
    revalidate: 1
  };
}
return {
  redirect: {
    permanent: false,
    destination: href
  },
  revalidate: 120
};
```

**Comment by stevenqzhang** [comment]
AFAICT this isn't in official documentation nor officially supported but it still works for me on Next 13!

**Reply by dimitriBouteille** [reply]
Work also for me, thanks :heart:

**Reply by CONSUMERTHOUGHTS** [reply]
Life saver thanks

**Reply by tuanpq1998** [reply]
Wow, there are many hidden things beyond the official documentation
